src/pages/api/secure/send-verification-code.ts

```diff
@@ -0,0 +1,54 @@
+import { createHmac, randomBytes, randomInt } from 'crypto'
+import { NextApiRequest, NextApiResponse } from 'next'
+
+import { withSessionRoute } from '@/ironAuth/withSessionApiRoute'
+import { sendVerificationCodeEmail } from '@/utils/email_helper'
+
+function hmacSha256(value: string, salt: string): string {
```

Comment:
node:crypto is the worst possible choice we can make for password hashes. Brute forcing pins, which are just 5 digits, would be easy. Replace all pin hashes with argon2 and add a salt secret as an ENV variable, so you save. PIN + SALT. This salt should be a 32-bit hex.